// @routerprotocol/chain-api/routerchain/voyager/funds_paid_request_pb.js
// source: routerchain/voyager/funds_paid_request.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var multichain_chain_type_pb = require('../../multichain/chain_type_pb.js');
goog.object.extend(proto, multichain_chain_type_pb);
var cosmos_base_v1beta1_coin_pb = require('../../cosmos/base/v1beta1/coin_pb.js');
goog.object.extend(proto, cosmos_base_v1beta1_coin_pb);
var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.FundsPaidRequest', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.FundsPaidRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.FundsPaidRequest.displayName = 'proto.routerprotocol.routerchain.voyager.FundsPaidRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.displayName = 'proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.FundsPaidRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    srcchainid: jspb.Message.getFieldWithDefault(msg, 1, ""),
    srcchaintype: jspb.Message.getFieldWithDefault(msg, 2, 0),
    srctxhash: jspb.Message.getFieldWithDefault(msg, 3, ""),
    srctimestamp: jspb.Message.getFieldWithDefault(msg, 4, 0),
    contract: jspb.Message.getFieldWithDefault(msg, 5, ""),
    eventnonce: jspb.Message.getFieldWithDefault(msg, 6, 0),
    blockheight: jspb.Message.getFieldWithDefault(msg, 7, 0),
    messagehash: msg.getMessagehash_asB64(),
    forwarder: jspb.Message.getFieldWithDefault(msg, 9, ""),
    forwarderrouteraddr: msg.getForwarderrouteraddr_asB64(),
    status: jspb.Message.getFieldWithDefault(msg, 11, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.FundsPaidRequest;
  return proto.routerprotocol.routerchain.voyager.FundsPaidRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrcchainid(value);
      break;
    case 2:
      var value = /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (reader.readEnum());
      msg.setSrcchaintype(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrctxhash(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setSrctimestamp(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setContract(value);
      break;
    case 6:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventnonce(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBlockheight(value);
      break;
    case 8:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setMessagehash(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.setForwarder(value);
      break;
    case 10:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setForwarderrouteraddr(value);
      break;
    case 11:
      var value = /** @type {string} */ (reader.readString());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.FundsPaidRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSrcchainid();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSrcchaintype();
  if (f !== 0.0) {
    writer.writeEnum(
      2,
      f
    );
  }
  f = message.getSrctxhash();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getSrctimestamp();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getContract();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getEventnonce();
  if (f !== 0) {
    writer.writeUint64(
      6,
      f
    );
  }
  f = message.getBlockheight();
  if (f !== 0) {
    writer.writeUint64(
      7,
      f
    );
  }
  f = message.getMessagehash_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      8,
      f
    );
  }
  f = message.getForwarder();
  if (f.length > 0) {
    writer.writeString(
      9,
      f
    );
  }
  f = message.getForwarderrouteraddr_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      10,
      f
    );
  }
  f = message.getStatus();
  if (f.length > 0) {
    writer.writeString(
      11,
      f
    );
  }
};


/**
 * optional string srcChainId = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getSrcchainid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setSrcchainid = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional routerprotocol.routerchain.multichain.ChainType srcChainType = 2;
 * @return {!proto.routerprotocol.routerchain.multichain.ChainType}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getSrcchaintype = function() {
  return /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {!proto.routerprotocol.routerchain.multichain.ChainType} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setSrcchaintype = function(value) {
  return jspb.Message.setProto3EnumField(this, 2, value);
};


/**
 * optional string srcTxHash = 3;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getSrctxhash = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setSrctxhash = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional uint64 srcTimestamp = 4;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getSrctimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setSrctimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string contract = 5;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setContract = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional uint64 eventNonce = 6;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getEventnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 6, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setEventnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 6, value);
};


/**
 * optional uint64 blockHeight = 7;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getBlockheight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setBlockheight = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


/**
 * optional bytes messageHash = 8;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getMessagehash = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * optional bytes messageHash = 8;
 * This is a type-conversion wrapper around `getMessagehash()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getMessagehash_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getMessagehash()));
};


/**
 * optional bytes messageHash = 8;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getMessagehash()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getMessagehash_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getMessagehash()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setMessagehash = function(value) {
  return jspb.Message.setProto3BytesField(this, 8, value);
};


/**
 * optional string forwarder = 9;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getForwarder = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setForwarder = function(value) {
  return jspb.Message.setProto3StringField(this, 9, value);
};


/**
 * optional bytes forwarderRouterAddr = 10;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getForwarderrouteraddr = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 10, ""));
};


/**
 * optional bytes forwarderRouterAddr = 10;
 * This is a type-conversion wrapper around `getForwarderrouteraddr()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getForwarderrouteraddr_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getForwarderrouteraddr()));
};


/**
 * optional bytes forwarderRouterAddr = 10;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getForwarderrouteraddr()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getForwarderrouteraddr_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getForwarderrouteraddr()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setForwarderrouteraddr = function(value) {
  return jspb.Message.setProto3BytesField(this, 10, value);
};


/**
 * optional string status = 11;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.getStatus = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 11, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequest.prototype.setStatus = function(value) {
  return jspb.Message.setProto3StringField(this, 11, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.toObject = function(includeInstance, msg) {
  var f, obj = {
    srcchainid: jspb.Message.getFieldWithDefault(msg, 1, ""),
    srcchaintype: jspb.Message.getFieldWithDefault(msg, 2, 0),
    srctxhash: jspb.Message.getFieldWithDefault(msg, 3, ""),
    srctimestamp: jspb.Message.getFieldWithDefault(msg, 4, 0),
    contract: jspb.Message.getFieldWithDefault(msg, 5, ""),
    eventnonce: jspb.Message.getFieldWithDefault(msg, 6, 0),
    blockheight: jspb.Message.getFieldWithDefault(msg, 7, 0),
    messagehash: msg.getMessagehash_asB64(),
    forwarder: jspb.Message.getFieldWithDefault(msg, 9, ""),
    forwarderrouteraddr: msg.getForwarderrouteraddr_asB64()
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash;
  return proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrcchainid(value);
      break;
    case 2:
      var value = /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (reader.readEnum());
      msg.setSrcchaintype(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrctxhash(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setSrctimestamp(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setContract(value);
      break;
    case 6:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventnonce(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBlockheight(value);
      break;
    case 8:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setMessagehash(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.setForwarder(value);
      break;
    case 10:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setForwarderrouteraddr(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSrcchainid();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSrcchaintype();
  if (f !== 0.0) {
    writer.writeEnum(
      2,
      f
    );
  }
  f = message.getSrctxhash();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getSrctimestamp();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getContract();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getEventnonce();
  if (f !== 0) {
    writer.writeUint64(
      6,
      f
    );
  }
  f = message.getBlockheight();
  if (f !== 0) {
    writer.writeUint64(
      7,
      f
    );
  }
  f = message.getMessagehash_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      8,
      f
    );
  }
  f = message.getForwarder();
  if (f.length > 0) {
    writer.writeString(
      9,
      f
    );
  }
  f = message.getForwarderrouteraddr_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      10,
      f
    );
  }
};


/**
 * optional string srcChainId = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getSrcchainid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setSrcchainid = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional routerprotocol.routerchain.multichain.ChainType srcChainType = 2;
 * @return {!proto.routerprotocol.routerchain.multichain.ChainType}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getSrcchaintype = function() {
  return /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {!proto.routerprotocol.routerchain.multichain.ChainType} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setSrcchaintype = function(value) {
  return jspb.Message.setProto3EnumField(this, 2, value);
};


/**
 * optional string srcTxHash = 3;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getSrctxhash = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setSrctxhash = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional uint64 srcTimestamp = 4;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getSrctimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setSrctimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string contract = 5;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setContract = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional uint64 eventNonce = 6;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getEventnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 6, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setEventnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 6, value);
};


/**
 * optional uint64 blockHeight = 7;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getBlockheight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setBlockheight = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


/**
 * optional bytes messageHash = 8;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getMessagehash = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * optional bytes messageHash = 8;
 * This is a type-conversion wrapper around `getMessagehash()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getMessagehash_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getMessagehash()));
};


/**
 * optional bytes messageHash = 8;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getMessagehash()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getMessagehash_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getMessagehash()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setMessagehash = function(value) {
  return jspb.Message.setProto3BytesField(this, 8, value);
};


/**
 * optional string forwarder = 9;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getForwarder = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setForwarder = function(value) {
  return jspb.Message.setProto3StringField(this, 9, value);
};


/**
 * optional bytes forwarderRouterAddr = 10;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getForwarderrouteraddr = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 10, ""));
};


/**
 * optional bytes forwarderRouterAddr = 10;
 * This is a type-conversion wrapper around `getForwarderrouteraddr()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getForwarderrouteraddr_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getForwarderrouteraddr()));
};


/**
 * optional bytes forwarderRouterAddr = 10;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getForwarderrouteraddr()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.getForwarderrouteraddr_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getForwarderrouteraddr()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash} returns this
 */
proto.routerprotocol.routerchain.voyager.FundsPaidRequestClaimHash.prototype.setForwarderrouteraddr = function(value) {
  return jspb.Message.setProto3BytesField(this, 10, value);
};


goog.object.extend(exports, proto.routerprotocol.routerchain.voyager);
