// @routerprotocol/chain-api/routerchain/voyager/genesis_pb.js
// source: routerchain/voyager/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var routerchain_voyager_params_pb = require('../../routerchain/voyager/params_pb.js');
goog.object.extend(proto, routerchain_voyager_params_pb);
var routerchain_voyager_fund_deposit_request_pb = require('../../routerchain/voyager/fund_deposit_request_pb.js');
goog.object.extend(proto, routerchain_voyager_fund_deposit_request_pb);
var routerchain_voyager_funds_paid_request_pb = require('../../routerchain/voyager/funds_paid_request_pb.js');
goog.object.extend(proto, routerchain_voyager_funds_paid_request_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.GenesisState', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.voyager.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.GenesisState.displayName = 'proto.routerprotocol.routerchain.voyager.GenesisState';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.voyager.GenesisState.repeatedFields_ = [2,3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && routerchain_voyager_params_pb.Params.toObject(includeInstance, f),
    funddepositrequestlistList: jspb.Message.toObjectList(msg.getFunddepositrequestlistList(),
    routerchain_voyager_fund_deposit_request_pb.FundDepositRequest.toObject, includeInstance),
    fundspaidrequestlistList: jspb.Message.toObjectList(msg.getFundspaidrequestlistList(),
    routerchain_voyager_funds_paid_request_pb.FundsPaidRequest.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.GenesisState;
  return proto.routerprotocol.routerchain.voyager.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new routerchain_voyager_params_pb.Params;
      reader.readMessage(value,routerchain_voyager_params_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = new routerchain_voyager_fund_deposit_request_pb.FundDepositRequest;
      reader.readMessage(value,routerchain_voyager_fund_deposit_request_pb.FundDepositRequest.deserializeBinaryFromReader);
      msg.addFunddepositrequestlist(value);
      break;
    case 3:
      var value = new routerchain_voyager_funds_paid_request_pb.FundsPaidRequest;
      reader.readMessage(value,routerchain_voyager_funds_paid_request_pb.FundsPaidRequest.deserializeBinaryFromReader);
      msg.addFundspaidrequestlist(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      routerchain_voyager_params_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getFunddepositrequestlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      routerchain_voyager_fund_deposit_request_pb.FundDepositRequest.serializeBinaryToWriter
    );
  }
  f = message.getFundspaidrequestlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      routerchain_voyager_funds_paid_request_pb.FundsPaidRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.routerprotocol.routerchain.voyager.Params}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.routerprotocol.routerchain.voyager.Params} */ (
    jspb.Message.getWrapperField(this, routerchain_voyager_params_pb.Params, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.voyager.Params|undefined} value
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState} returns this
*/
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState} returns this
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated FundDepositRequest fundDepositRequestList = 2;
 * @return {!Array<!proto.routerprotocol.routerchain.voyager.FundDepositRequest>}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.getFunddepositrequestlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.voyager.FundDepositRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_voyager_fund_deposit_request_pb.FundDepositRequest, 2));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.voyager.FundDepositRequest>} value
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState} returns this
*/
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.setFunddepositrequestlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.voyager.FundDepositRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.voyager.FundDepositRequest}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.addFunddepositrequestlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.routerprotocol.routerchain.voyager.FundDepositRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState} returns this
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.clearFunddepositrequestlistList = function() {
  return this.setFunddepositrequestlistList([]);
};


/**
 * repeated FundsPaidRequest fundsPaidRequestList = 3;
 * @return {!Array<!proto.routerprotocol.routerchain.voyager.FundsPaidRequest>}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.getFundspaidrequestlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.voyager.FundsPaidRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_voyager_funds_paid_request_pb.FundsPaidRequest, 3));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.voyager.FundsPaidRequest>} value
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState} returns this
*/
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.setFundspaidrequestlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.voyager.FundsPaidRequest}
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.addFundspaidrequestlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.routerprotocol.routerchain.voyager.FundsPaidRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.voyager.GenesisState} returns this
 */
proto.routerprotocol.routerchain.voyager.GenesisState.prototype.clearFundspaidrequestlistList = function() {
  return this.setFundspaidrequestlistList([]);
};


goog.object.extend(exports, proto.routerprotocol.routerchain.voyager);
