// @routerprotocol/chain-api/routerchain/voyager/tx_pb.js
// source: routerchain/voyager/tx.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var multichain_chain_type_pb = require('../../multichain/chain_type_pb.js');
goog.object.extend(proto, multichain_chain_type_pb);
var cosmos_base_v1beta1_coin_pb = require('../../cosmos/base/v1beta1/coin_pb.js');
goog.object.extend(proto, cosmos_base_v1beta1_coin_pb);
var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.MsgFundsDeposited', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.MsgFundsPaid', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.MsgFundsDeposited, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.displayName = 'proto.routerprotocol.routerchain.voyager.MsgFundsDeposited';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.displayName = 'proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.MsgFundsPaid, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.MsgFundsPaid.displayName = 'proto.routerprotocol.routerchain.voyager.MsgFundsPaid';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.displayName = 'proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.toObject = function(includeInstance, msg) {
  var f, obj = {
    orchestrator: jspb.Message.getFieldWithDefault(msg, 1, ""),
    srcchainid: jspb.Message.getFieldWithDefault(msg, 2, ""),
    srcchaintype: jspb.Message.getFieldWithDefault(msg, 3, 0),
    srctxhash: jspb.Message.getFieldWithDefault(msg, 4, ""),
    srctimestamp: jspb.Message.getFieldWithDefault(msg, 5, 0),
    contract: jspb.Message.getFieldWithDefault(msg, 6, ""),
    depositId: jspb.Message.getFieldWithDefault(msg, 7, 0),
    blockheight: jspb.Message.getFieldWithDefault(msg, 8, 0),
    destchainid: msg.getDestchainid_asB64(),
    amount: jspb.Message.getFieldWithDefault(msg, 10, ""),
    relayerfees: jspb.Message.getFieldWithDefault(msg, 11, ""),
    srctoken: jspb.Message.getFieldWithDefault(msg, 12, ""),
    recipient: msg.getRecipient_asB64(),
    depositor: jspb.Message.getFieldWithDefault(msg, 14, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.MsgFundsDeposited;
  return proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestrator(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrcchainid(value);
      break;
    case 3:
      var value = /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (reader.readEnum());
      msg.setSrcchaintype(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrctxhash(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setSrctimestamp(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setContract(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setDepositId(value);
      break;
    case 8:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBlockheight(value);
      break;
    case 9:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setDestchainid(value);
      break;
    case 10:
      var value = /** @type {string} */ (reader.readString());
      msg.setAmount(value);
      break;
    case 11:
      var value = /** @type {string} */ (reader.readString());
      msg.setRelayerfees(value);
      break;
    case 12:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrctoken(value);
      break;
    case 13:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setRecipient(value);
      break;
    case 14:
      var value = /** @type {string} */ (reader.readString());
      msg.setDepositor(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getOrchestrator();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSrcchainid();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getSrcchaintype();
  if (f !== 0.0) {
    writer.writeEnum(
      3,
      f
    );
  }
  f = message.getSrctxhash();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getSrctimestamp();
  if (f !== 0) {
    writer.writeUint64(
      5,
      f
    );
  }
  f = message.getContract();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getDepositId();
  if (f !== 0) {
    writer.writeUint64(
      7,
      f
    );
  }
  f = message.getBlockheight();
  if (f !== 0) {
    writer.writeUint64(
      8,
      f
    );
  }
  f = message.getDestchainid_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      9,
      f
    );
  }
  f = message.getAmount();
  if (f.length > 0) {
    writer.writeString(
      10,
      f
    );
  }
  f = message.getRelayerfees();
  if (f.length > 0) {
    writer.writeString(
      11,
      f
    );
  }
  f = message.getSrctoken();
  if (f.length > 0) {
    writer.writeString(
      12,
      f
    );
  }
  f = message.getRecipient_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      13,
      f
    );
  }
  f = message.getDepositor();
  if (f.length > 0) {
    writer.writeString(
      14,
      f
    );
  }
};


/**
 * optional string orchestrator = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getOrchestrator = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setOrchestrator = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string srcChainId = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getSrcchainid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setSrcchainid = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional routerprotocol.routerchain.multichain.ChainType srcChainType = 3;
 * @return {!proto.routerprotocol.routerchain.multichain.ChainType}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getSrcchaintype = function() {
  return /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {!proto.routerprotocol.routerchain.multichain.ChainType} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setSrcchaintype = function(value) {
  return jspb.Message.setProto3EnumField(this, 3, value);
};


/**
 * optional string srcTxHash = 4;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getSrctxhash = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setSrctxhash = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional uint64 srcTimestamp = 5;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getSrctimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setSrctimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};


/**
 * optional string contract = 6;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setContract = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional uint64 deposit_id = 7;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getDepositId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setDepositId = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


/**
 * optional uint64 blockHeight = 8;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getBlockheight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setBlockheight = function(value) {
  return jspb.Message.setProto3IntField(this, 8, value);
};


/**
 * optional bytes destChainId = 9;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getDestchainid = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * optional bytes destChainId = 9;
 * This is a type-conversion wrapper around `getDestchainid()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getDestchainid_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getDestchainid()));
};


/**
 * optional bytes destChainId = 9;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getDestchainid()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getDestchainid_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getDestchainid()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setDestchainid = function(value) {
  return jspb.Message.setProto3BytesField(this, 9, value);
};


/**
 * optional string amount = 10;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getAmount = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 10, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setAmount = function(value) {
  return jspb.Message.setProto3StringField(this, 10, value);
};


/**
 * optional string relayerFees = 11;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getRelayerfees = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 11, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setRelayerfees = function(value) {
  return jspb.Message.setProto3StringField(this, 11, value);
};


/**
 * optional string srcToken = 12;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getSrctoken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 12, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setSrctoken = function(value) {
  return jspb.Message.setProto3StringField(this, 12, value);
};


/**
 * optional bytes recipient = 13;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getRecipient = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 13, ""));
};


/**
 * optional bytes recipient = 13;
 * This is a type-conversion wrapper around `getRecipient()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getRecipient_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getRecipient()));
};


/**
 * optional bytes recipient = 13;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getRecipient()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getRecipient_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getRecipient()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setRecipient = function(value) {
  return jspb.Message.setProto3BytesField(this, 13, value);
};


/**
 * optional string depositor = 14;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.getDepositor = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 14, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDeposited} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDeposited.prototype.setDepositor = function(value) {
  return jspb.Message.setProto3StringField(this, 14, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse;
  return proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsDepositedResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.MsgFundsPaid.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.toObject = function(includeInstance, msg) {
  var f, obj = {
    orchestrator: jspb.Message.getFieldWithDefault(msg, 1, ""),
    srcchainid: jspb.Message.getFieldWithDefault(msg, 2, ""),
    srcchaintype: jspb.Message.getFieldWithDefault(msg, 3, 0),
    srctxhash: jspb.Message.getFieldWithDefault(msg, 4, ""),
    srctimestamp: jspb.Message.getFieldWithDefault(msg, 5, 0),
    contract: jspb.Message.getFieldWithDefault(msg, 6, ""),
    eventnonce: jspb.Message.getFieldWithDefault(msg, 7, 0),
    blockheight: jspb.Message.getFieldWithDefault(msg, 8, 0),
    messagehash: msg.getMessagehash_asB64(),
    forwarder: jspb.Message.getFieldWithDefault(msg, 10, ""),
    forwarderrouteraddr: msg.getForwarderrouteraddr_asB64()
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.MsgFundsPaid;
  return proto.routerprotocol.routerchain.voyager.MsgFundsPaid.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrchestrator(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrcchainid(value);
      break;
    case 3:
      var value = /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (reader.readEnum());
      msg.setSrcchaintype(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setSrctxhash(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setSrctimestamp(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setContract(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventnonce(value);
      break;
    case 8:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBlockheight(value);
      break;
    case 9:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setMessagehash(value);
      break;
    case 10:
      var value = /** @type {string} */ (reader.readString());
      msg.setForwarder(value);
      break;
    case 11:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setForwarderrouteraddr(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.MsgFundsPaid.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getOrchestrator();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSrcchainid();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getSrcchaintype();
  if (f !== 0.0) {
    writer.writeEnum(
      3,
      f
    );
  }
  f = message.getSrctxhash();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getSrctimestamp();
  if (f !== 0) {
    writer.writeUint64(
      5,
      f
    );
  }
  f = message.getContract();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getEventnonce();
  if (f !== 0) {
    writer.writeUint64(
      7,
      f
    );
  }
  f = message.getBlockheight();
  if (f !== 0) {
    writer.writeUint64(
      8,
      f
    );
  }
  f = message.getMessagehash_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      9,
      f
    );
  }
  f = message.getForwarder();
  if (f.length > 0) {
    writer.writeString(
      10,
      f
    );
  }
  f = message.getForwarderrouteraddr_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      11,
      f
    );
  }
};


/**
 * optional string orchestrator = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getOrchestrator = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setOrchestrator = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string srcChainId = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getSrcchainid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setSrcchainid = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional routerprotocol.routerchain.multichain.ChainType srcChainType = 3;
 * @return {!proto.routerprotocol.routerchain.multichain.ChainType}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getSrcchaintype = function() {
  return /** @type {!proto.routerprotocol.routerchain.multichain.ChainType} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {!proto.routerprotocol.routerchain.multichain.ChainType} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setSrcchaintype = function(value) {
  return jspb.Message.setProto3EnumField(this, 3, value);
};


/**
 * optional string srcTxHash = 4;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getSrctxhash = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setSrctxhash = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional uint64 srcTimestamp = 5;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getSrctimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setSrctimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};


/**
 * optional string contract = 6;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setContract = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional uint64 eventNonce = 7;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getEventnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setEventnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


/**
 * optional uint64 blockHeight = 8;
 * @return {number}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getBlockheight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setBlockheight = function(value) {
  return jspb.Message.setProto3IntField(this, 8, value);
};


/**
 * optional bytes messageHash = 9;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getMessagehash = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * optional bytes messageHash = 9;
 * This is a type-conversion wrapper around `getMessagehash()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getMessagehash_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getMessagehash()));
};


/**
 * optional bytes messageHash = 9;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getMessagehash()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getMessagehash_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getMessagehash()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setMessagehash = function(value) {
  return jspb.Message.setProto3BytesField(this, 9, value);
};


/**
 * optional string forwarder = 10;
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getForwarder = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 10, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setForwarder = function(value) {
  return jspb.Message.setProto3StringField(this, 10, value);
};


/**
 * optional bytes forwarderRouterAddr = 11;
 * @return {!(string|Uint8Array)}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getForwarderrouteraddr = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 11, ""));
};


/**
 * optional bytes forwarderRouterAddr = 11;
 * This is a type-conversion wrapper around `getForwarderrouteraddr()`
 * @return {string}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getForwarderrouteraddr_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getForwarderrouteraddr()));
};


/**
 * optional bytes forwarderRouterAddr = 11;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getForwarderrouteraddr()`
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.getForwarderrouteraddr_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getForwarderrouteraddr()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaid} returns this
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaid.prototype.setForwarderrouteraddr = function(value) {
  return jspb.Message.setProto3BytesField(this, 11, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse;
  return proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.voyager.MsgFundsPaidResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};


goog.object.extend(exports, proto.routerprotocol.routerchain.voyager);
