// @routerprotocol/chain-api/routerchain/voyager/tx_pb_service.js
// package: routerprotocol.routerchain.voyager
// file: routerchain/voyager/tx.proto

var routerchain_voyager_tx_pb = require("../../routerchain/voyager/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "routerprotocol.routerchain.voyager.Msg";
  return Msg;
}());

Msg.FundsDeposited = {
  methodName: "FundsDeposited",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_voyager_tx_pb.MsgFundsDeposited,
  responseType: routerchain_voyager_tx_pb.MsgFundsDepositedResponse
};

Msg.FundsPaid = {
  methodName: "FundsPaid",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_voyager_tx_pb.MsgFundsPaid,
  responseType: routerchain_voyager_tx_pb.MsgFundsPaidResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.fundsDeposited = function fundsDeposited(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.FundsDeposited, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.fundsPaid = function fundsPaid(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.FundsPaid, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
