// @routerprotocol/chain-api/routerchain/crosschain/crosschain_tx_status_pb.js
// source: routerchain/crosschain/crosschain_tx_status.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var google_protobuf_any_pb = require('google-protobuf/google/protobuf/any_pb.js');
goog.object.extend(proto, google_protobuf_any_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.crosschain.CrosschainTxStatus', null, global);
/**
 * @enum {number}
 */
proto.routerprotocol.routerchain.crosschain.CrosschainTxStatus = {
  CROSSCHAIN_TX_CREATED: 0,
  CROSSCHAIN_TX_VALIDATED: 1,
  NATIVE_TOKEN_TRANFERRED: 2,
  CROSSCHAIN_TX_READY_TO_EXECUTE: 3,
  CROSSCHAIN_TX_EXECUTED: 4,
  CROSSCHAIN_TX_EXECUTION_FAILED: 5,
  CROSSCHAIN_TX_FEES_SETTLED: 6,
  CROSSCHAIN_TX_COMPLETED: 7
};

goog.object.extend(exports, proto.routerprotocol.routerchain.crosschain);
