// @routerprotocol/chain-api/routerchain/crosschain/genesis_pb.js
// source: routerchain/crosschain/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var routerchain_crosschain_params_pb = require('../../routerchain/crosschain/params_pb.js');
goog.object.extend(proto, routerchain_crosschain_params_pb);
var routerchain_crosschain_crosschain_request_pb = require('../../routerchain/crosschain/crosschain_request_pb.js');
goog.object.extend(proto, routerchain_crosschain_crosschain_request_pb);
var routerchain_crosschain_crosschain_request_confirm_pb = require('../../routerchain/crosschain/crosschain_request_confirm_pb.js');
goog.object.extend(proto, routerchain_crosschain_crosschain_request_confirm_pb);
var routerchain_crosschain_crosschain_ack_request_pb = require('../../routerchain/crosschain/crosschain_ack_request_pb.js');
goog.object.extend(proto, routerchain_crosschain_crosschain_ack_request_pb);
var routerchain_crosschain_crosschain_ack_request_confirm_pb = require('../../routerchain/crosschain/crosschain_ack_request_confirm_pb.js');
goog.object.extend(proto, routerchain_crosschain_crosschain_ack_request_confirm_pb);
var routerchain_crosschain_crosschain_ack_receipt_pb = require('../../routerchain/crosschain/crosschain_ack_receipt_pb.js');
goog.object.extend(proto, routerchain_crosschain_crosschain_ack_receipt_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.crosschain.GenesisState', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.crosschain.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.crosschain.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.crosschain.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.crosschain.GenesisState.displayName = 'proto.routerprotocol.routerchain.crosschain.GenesisState';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.repeatedFields_ = [2,3,4,5,6];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.crosschain.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.crosschain.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && routerchain_crosschain_params_pb.Params.toObject(includeInstance, f),
    crosschainrequestlistList: jspb.Message.toObjectList(msg.getCrosschainrequestlistList(),
    routerchain_crosschain_crosschain_request_pb.CrosschainRequest.toObject, includeInstance),
    crosschainrequestconfirmlistList: jspb.Message.toObjectList(msg.getCrosschainrequestconfirmlistList(),
    routerchain_crosschain_crosschain_request_confirm_pb.CrosschainRequestConfirm.toObject, includeInstance),
    crosschainackrequestlistList: jspb.Message.toObjectList(msg.getCrosschainackrequestlistList(),
    routerchain_crosschain_crosschain_ack_request_pb.CrosschainAckRequest.toObject, includeInstance),
    crosschainackrequestconfirmlistList: jspb.Message.toObjectList(msg.getCrosschainackrequestconfirmlistList(),
    routerchain_crosschain_crosschain_ack_request_confirm_pb.CrosschainAckRequestConfirm.toObject, includeInstance),
    crosschainackreceiptlistList: jspb.Message.toObjectList(msg.getCrosschainackreceiptlistList(),
    routerchain_crosschain_crosschain_ack_receipt_pb.CrosschainAckReceipt.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.crosschain.GenesisState;
  return proto.routerprotocol.routerchain.crosschain.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.crosschain.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new routerchain_crosschain_params_pb.Params;
      reader.readMessage(value,routerchain_crosschain_params_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = new routerchain_crosschain_crosschain_request_pb.CrosschainRequest;
      reader.readMessage(value,routerchain_crosschain_crosschain_request_pb.CrosschainRequest.deserializeBinaryFromReader);
      msg.addCrosschainrequestlist(value);
      break;
    case 3:
      var value = new routerchain_crosschain_crosschain_request_confirm_pb.CrosschainRequestConfirm;
      reader.readMessage(value,routerchain_crosschain_crosschain_request_confirm_pb.CrosschainRequestConfirm.deserializeBinaryFromReader);
      msg.addCrosschainrequestconfirmlist(value);
      break;
    case 4:
      var value = new routerchain_crosschain_crosschain_ack_request_pb.CrosschainAckRequest;
      reader.readMessage(value,routerchain_crosschain_crosschain_ack_request_pb.CrosschainAckRequest.deserializeBinaryFromReader);
      msg.addCrosschainackrequestlist(value);
      break;
    case 5:
      var value = new routerchain_crosschain_crosschain_ack_request_confirm_pb.CrosschainAckRequestConfirm;
      reader.readMessage(value,routerchain_crosschain_crosschain_ack_request_confirm_pb.CrosschainAckRequestConfirm.deserializeBinaryFromReader);
      msg.addCrosschainackrequestconfirmlist(value);
      break;
    case 6:
      var value = new routerchain_crosschain_crosschain_ack_receipt_pb.CrosschainAckReceipt;
      reader.readMessage(value,routerchain_crosschain_crosschain_ack_receipt_pb.CrosschainAckReceipt.deserializeBinaryFromReader);
      msg.addCrosschainackreceiptlist(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.crosschain.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.crosschain.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      routerchain_crosschain_params_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getCrosschainrequestlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      routerchain_crosschain_crosschain_request_pb.CrosschainRequest.serializeBinaryToWriter
    );
  }
  f = message.getCrosschainrequestconfirmlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      routerchain_crosschain_crosschain_request_confirm_pb.CrosschainRequestConfirm.serializeBinaryToWriter
    );
  }
  f = message.getCrosschainackrequestlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      routerchain_crosschain_crosschain_ack_request_pb.CrosschainAckRequest.serializeBinaryToWriter
    );
  }
  f = message.getCrosschainackrequestconfirmlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      routerchain_crosschain_crosschain_ack_request_confirm_pb.CrosschainAckRequestConfirm.serializeBinaryToWriter
    );
  }
  f = message.getCrosschainackreceiptlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      6,
      f,
      routerchain_crosschain_crosschain_ack_receipt_pb.CrosschainAckReceipt.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.routerprotocol.routerchain.crosschain.Params}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.routerprotocol.routerchain.crosschain.Params} */ (
    jspb.Message.getWrapperField(this, routerchain_crosschain_params_pb.Params, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.crosschain.Params|undefined} value
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
*/
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated CrosschainRequest crosschainRequestList = 2;
 * @return {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainRequest>}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.getCrosschainrequestlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_crosschain_crosschain_request_pb.CrosschainRequest, 2));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainRequest>} value
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
*/
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.setCrosschainrequestlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.crosschain.CrosschainRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.crosschain.CrosschainRequest}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.addCrosschainrequestlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.routerprotocol.routerchain.crosschain.CrosschainRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.clearCrosschainrequestlistList = function() {
  return this.setCrosschainrequestlistList([]);
};


/**
 * repeated CrosschainRequestConfirm crosschainRequestConfirmList = 3;
 * @return {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainRequestConfirm>}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.getCrosschainrequestconfirmlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainRequestConfirm>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_crosschain_crosschain_request_confirm_pb.CrosschainRequestConfirm, 3));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainRequestConfirm>} value
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
*/
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.setCrosschainrequestconfirmlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.crosschain.CrosschainRequestConfirm=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.crosschain.CrosschainRequestConfirm}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.addCrosschainrequestconfirmlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.routerprotocol.routerchain.crosschain.CrosschainRequestConfirm, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.clearCrosschainrequestconfirmlistList = function() {
  return this.setCrosschainrequestconfirmlistList([]);
};


/**
 * repeated CrosschainAckRequest crosschainAckRequestList = 4;
 * @return {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequest>}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.getCrosschainackrequestlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_crosschain_crosschain_ack_request_pb.CrosschainAckRequest, 4));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequest>} value
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
*/
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.setCrosschainackrequestlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequest}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.addCrosschainackrequestlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.routerprotocol.routerchain.crosschain.CrosschainAckRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.clearCrosschainackrequestlistList = function() {
  return this.setCrosschainackrequestlistList([]);
};


/**
 * repeated CrosschainAckRequestConfirm crosschainAckRequestConfirmList = 5;
 * @return {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequestConfirm>}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.getCrosschainackrequestconfirmlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequestConfirm>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_crosschain_crosschain_ack_request_confirm_pb.CrosschainAckRequestConfirm, 5));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequestConfirm>} value
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
*/
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.setCrosschainackrequestconfirmlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequestConfirm=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.crosschain.CrosschainAckRequestConfirm}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.addCrosschainackrequestconfirmlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.routerprotocol.routerchain.crosschain.CrosschainAckRequestConfirm, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.clearCrosschainackrequestconfirmlistList = function() {
  return this.setCrosschainackrequestconfirmlistList([]);
};


/**
 * repeated CrosschainAckReceipt crosschainAckReceiptList = 6;
 * @return {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckReceipt>}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.getCrosschainackreceiptlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckReceipt>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_crosschain_crosschain_ack_receipt_pb.CrosschainAckReceipt, 6));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.crosschain.CrosschainAckReceipt>} value
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
*/
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.setCrosschainackreceiptlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 6, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.crosschain.CrosschainAckReceipt=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.crosschain.CrosschainAckReceipt}
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.addCrosschainackreceiptlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 6, opt_value, proto.routerprotocol.routerchain.crosschain.CrosschainAckReceipt, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.crosschain.GenesisState} returns this
 */
proto.routerprotocol.routerchain.crosschain.GenesisState.prototype.clearCrosschainackreceiptlistList = function() {
  return this.setCrosschainackreceiptlistList([]);
};


goog.object.extend(exports, proto.routerprotocol.routerchain.crosschain);
