// @routerprotocol/chain-api/routerchain/crosschain/tx_pb_service.js
// package: routerprotocol.routerchain.crosschain
// file: routerchain/crosschain/tx.proto

var routerchain_crosschain_tx_pb = require("../../routerchain/crosschain/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "routerprotocol.routerchain.crosschain.Msg";
  return Msg;
}());

Msg.CrosschainRequest = {
  methodName: "CrosschainRequest",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_crosschain_tx_pb.MsgCrosschainRequest,
  responseType: routerchain_crosschain_tx_pb.MsgCrosschainRequestResponse
};

Msg.ConfirmCrosschainRequest = {
  methodName: "ConfirmCrosschainRequest",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_crosschain_tx_pb.MsgConfirmCrosschainRequest,
  responseType: routerchain_crosschain_tx_pb.MsgConfirmCrosschainRequestResponse
};

Msg.CrosschainAckRequest = {
  methodName: "CrosschainAckRequest",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_crosschain_tx_pb.MsgCrosschainAckRequest,
  responseType: routerchain_crosschain_tx_pb.MsgCrosschainAckRequestResponse
};

Msg.ConfirmCrosschainAckRequest = {
  methodName: "ConfirmCrosschainAckRequest",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_crosschain_tx_pb.MsgConfirmCrosschainAckRequest,
  responseType: routerchain_crosschain_tx_pb.MsgConfirmCrosschainAckRequestResponse
};

Msg.CrosschainAckReceipt = {
  methodName: "CrosschainAckReceipt",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_crosschain_tx_pb.MsgCrosschainAckReceipt,
  responseType: routerchain_crosschain_tx_pb.MsgCrosschainAckReceiptResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.crosschainRequest = function crosschainRequest(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.CrosschainRequest, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.confirmCrosschainRequest = function confirmCrosschainRequest(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.ConfirmCrosschainRequest, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.crosschainAckRequest = function crosschainAckRequest(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.CrosschainAckRequest, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.confirmCrosschainAckRequest = function confirmCrosschainAckRequest(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.ConfirmCrosschainAckRequest, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.crosschainAckReceipt = function crosschainAckReceipt(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.CrosschainAckReceipt, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
