// @routerprotocol/chain-api/routerchain/crosschain/workflow_type_pb.js
// source: routerchain/crosschain/workflow_type.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var google_protobuf_any_pb = require('google-protobuf/google/protobuf/any_pb.js');
goog.object.extend(proto, google_protobuf_any_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.crosschain.WorkflowType', null, global);
/**
 * @enum {number}
 */
proto.routerprotocol.routerchain.crosschain.WorkflowType = {
  NONE: 0,
  INBOUND: 1,
  CROSSTALK: 2,
  OUTBOUND: 3,
  INBOUND_ACK: 4,
  CROSSTALK_ACK: 5,
  OUTBOUND_ACK: 6
};

goog.object.extend(exports, proto.routerprotocol.routerchain.crosschain);
