// @routerprotocol/chain-api/routerchain/pricefeed/genesis_pb.js
// source: routerchain/pricefeed/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var routerchain_pricefeed_params_pb = require('../../routerchain/pricefeed/params_pb.js');
goog.object.extend(proto, routerchain_pricefeed_params_pb);
var routerchain_pricefeed_price_pb = require('../../routerchain/pricefeed/price_pb.js');
goog.object.extend(proto, routerchain_pricefeed_price_pb);
var routerchain_pricefeed_symbol_request_pb = require('../../routerchain/pricefeed/symbol_request_pb.js');
goog.object.extend(proto, routerchain_pricefeed_symbol_request_pb);
var routerchain_pricefeed_price_feeder_info_pb = require('../../routerchain/pricefeed/price_feeder_info_pb.js');
goog.object.extend(proto, routerchain_pricefeed_price_feeder_info_pb);
var routerchain_pricefeed_gas_price_pb = require('../../routerchain/pricefeed/gas_price_pb.js');
goog.object.extend(proto, routerchain_pricefeed_gas_price_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.pricefeed.GenesisState', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.pricefeed.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.pricefeed.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.pricefeed.GenesisState.displayName = 'proto.routerprotocol.routerchain.pricefeed.GenesisState';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.repeatedFields_ = [3,4,5,6];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.pricefeed.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.pricefeed.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && routerchain_pricefeed_params_pb.Params.toObject(includeInstance, f),
    portId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    pricelistList: jspb.Message.toObjectList(msg.getPricelistList(),
    routerchain_pricefeed_price_pb.Price.toObject, includeInstance),
    symbolrequestlistList: jspb.Message.toObjectList(msg.getSymbolrequestlistList(),
    routerchain_pricefeed_symbol_request_pb.SymbolRequest.toObject, includeInstance),
    pricefeederinfolistList: jspb.Message.toObjectList(msg.getPricefeederinfolistList(),
    routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo.toObject, includeInstance),
    gaspricelistList: jspb.Message.toObjectList(msg.getGaspricelistList(),
    routerchain_pricefeed_gas_price_pb.GasPrice.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.pricefeed.GenesisState;
  return proto.routerprotocol.routerchain.pricefeed.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.pricefeed.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new routerchain_pricefeed_params_pb.Params;
      reader.readMessage(value,routerchain_pricefeed_params_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setPortId(value);
      break;
    case 3:
      var value = new routerchain_pricefeed_price_pb.Price;
      reader.readMessage(value,routerchain_pricefeed_price_pb.Price.deserializeBinaryFromReader);
      msg.addPricelist(value);
      break;
    case 4:
      var value = new routerchain_pricefeed_symbol_request_pb.SymbolRequest;
      reader.readMessage(value,routerchain_pricefeed_symbol_request_pb.SymbolRequest.deserializeBinaryFromReader);
      msg.addSymbolrequestlist(value);
      break;
    case 5:
      var value = new routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo;
      reader.readMessage(value,routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo.deserializeBinaryFromReader);
      msg.addPricefeederinfolist(value);
      break;
    case 6:
      var value = new routerchain_pricefeed_gas_price_pb.GasPrice;
      reader.readMessage(value,routerchain_pricefeed_gas_price_pb.GasPrice.deserializeBinaryFromReader);
      msg.addGaspricelist(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.pricefeed.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.pricefeed.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      routerchain_pricefeed_params_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getPortId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getPricelistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      routerchain_pricefeed_price_pb.Price.serializeBinaryToWriter
    );
  }
  f = message.getSymbolrequestlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      routerchain_pricefeed_symbol_request_pb.SymbolRequest.serializeBinaryToWriter
    );
  }
  f = message.getPricefeederinfolistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo.serializeBinaryToWriter
    );
  }
  f = message.getGaspricelistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      6,
      f,
      routerchain_pricefeed_gas_price_pb.GasPrice.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.routerprotocol.routerchain.pricefeed.Params}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.routerprotocol.routerchain.pricefeed.Params} */ (
    jspb.Message.getWrapperField(this, routerchain_pricefeed_params_pb.Params, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.pricefeed.Params|undefined} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
*/
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string port_id = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.getPortId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.setPortId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * repeated Price priceList = 3;
 * @return {!Array<!proto.routerprotocol.routerchain.pricefeed.Price>}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.getPricelistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.pricefeed.Price>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_pricefeed_price_pb.Price, 3));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.pricefeed.Price>} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
*/
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.setPricelistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.pricefeed.Price=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.pricefeed.Price}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.addPricelist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.routerprotocol.routerchain.pricefeed.Price, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.clearPricelistList = function() {
  return this.setPricelistList([]);
};


/**
 * repeated SymbolRequest symbolRequestList = 4;
 * @return {!Array<!proto.routerprotocol.routerchain.pricefeed.SymbolRequest>}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.getSymbolrequestlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.pricefeed.SymbolRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_pricefeed_symbol_request_pb.SymbolRequest, 4));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.pricefeed.SymbolRequest>} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
*/
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.setSymbolrequestlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.pricefeed.SymbolRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.pricefeed.SymbolRequest}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.addSymbolrequestlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.routerprotocol.routerchain.pricefeed.SymbolRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.clearSymbolrequestlistList = function() {
  return this.setSymbolrequestlistList([]);
};


/**
 * repeated PriceFeederInfo priceFeederInfoList = 5;
 * @return {!Array<!proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo>}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.getPricefeederinfolistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo, 5));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo>} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
*/
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.setPricefeederinfolistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.addPricefeederinfolist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.clearPricefeederinfolistList = function() {
  return this.setPricefeederinfolistList([]);
};


/**
 * repeated GasPrice gasPriceList = 6;
 * @return {!Array<!proto.routerprotocol.routerchain.pricefeed.GasPrice>}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.getGaspricelistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.pricefeed.GasPrice>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_pricefeed_gas_price_pb.GasPrice, 6));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.pricefeed.GasPrice>} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
*/
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.setGaspricelistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 6, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.pricefeed.GasPrice=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.pricefeed.GasPrice}
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.addGaspricelist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 6, opt_value, proto.routerprotocol.routerchain.pricefeed.GasPrice, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.pricefeed.GenesisState} returns this
 */
proto.routerprotocol.routerchain.pricefeed.GenesisState.prototype.clearGaspricelistList = function() {
  return this.setGaspricelistList([]);
};


goog.object.extend(exports, proto.routerprotocol.routerchain.pricefeed);
