// @routerprotocol/chain-api/routerchain/pricefeed/tx_pb_service.js
// package: routerprotocol.routerchain.pricefeed
// file: routerchain/pricefeed/tx.proto

var routerchain_pricefeed_tx_pb = require("../../routerchain/pricefeed/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "routerprotocol.routerchain.pricefeed.Msg";
  return Msg;
}());

Msg.TokenPrices = {
  methodName: "TokenPrices",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_pricefeed_tx_pb.MsgTokenPrices,
  responseType: routerchain_pricefeed_tx_pb.MsgTokenPricesResponse
};

Msg.GasPrices = {
  methodName: "GasPrices",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: routerchain_pricefeed_tx_pb.MsgGasPrices,
  responseType: routerchain_pricefeed_tx_pb.MsgGasPricesResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.tokenPrices = function tokenPrices(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.TokenPrices, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.gasPrices = function gasPrices(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.GasPrices, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
