// @routerprotocol/chain-api/routerchain/pricefeed/tx_pb.js
// source: routerchain/pricefeed/tx.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var routerchain_pricefeed_price_pb = require('../../routerchain/pricefeed/price_pb.js');
goog.object.extend(proto, routerchain_pricefeed_price_pb);
var routerchain_pricefeed_gas_price_pb = require('../../routerchain/pricefeed/gas_price_pb.js');
goog.object.extend(proto, routerchain_pricefeed_gas_price_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.pricefeed.MsgGasPrices', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.displayName = 'proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.displayName = 'proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.pricefeed.MsgGasPrices, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.displayName = 'proto.routerprotocol.routerchain.pricefeed.MsgGasPrices';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.displayName = 'proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.toObject = function(includeInstance, msg) {
  var f, obj = {
    priceFeederAddress: jspb.Message.getFieldWithDefault(msg, 1, ""),
    priceFeederName: jspb.Message.getFieldWithDefault(msg, 2, ""),
    tokenPricesList: jspb.Message.toObjectList(msg.getTokenPricesList(),
    routerchain_pricefeed_price_pb.Price.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices;
  return proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setPriceFeederAddress(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setPriceFeederName(value);
      break;
    case 3:
      var value = new routerchain_pricefeed_price_pb.Price;
      reader.readMessage(value,routerchain_pricefeed_price_pb.Price.deserializeBinaryFromReader);
      msg.addTokenPrices(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPriceFeederAddress();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPriceFeederName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getTokenPricesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      routerchain_pricefeed_price_pb.Price.serializeBinaryToWriter
    );
  }
};


/**
 * optional string price_feeder_address = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.getPriceFeederAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} returns this
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.setPriceFeederAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string price_feeder_name = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.getPriceFeederName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} returns this
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.setPriceFeederName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * repeated Price token_prices = 3;
 * @return {!Array<!proto.routerprotocol.routerchain.pricefeed.Price>}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.getTokenPricesList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.pricefeed.Price>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_pricefeed_price_pb.Price, 3));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.pricefeed.Price>} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} returns this
*/
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.setTokenPricesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.pricefeed.Price=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.pricefeed.Price}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.addTokenPrices = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.routerprotocol.routerchain.pricefeed.Price, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices} returns this
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPrices.prototype.clearTokenPricesList = function() {
  return this.setTokenPricesList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse;
  return proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgTokenPricesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.toObject = function(includeInstance, msg) {
  var f, obj = {
    priceFeederAddress: jspb.Message.getFieldWithDefault(msg, 1, ""),
    priceFeederName: jspb.Message.getFieldWithDefault(msg, 2, ""),
    gasPricesList: jspb.Message.toObjectList(msg.getGasPricesList(),
    routerchain_pricefeed_gas_price_pb.GasPrice.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.pricefeed.MsgGasPrices;
  return proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setPriceFeederAddress(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setPriceFeederName(value);
      break;
    case 3:
      var value = new routerchain_pricefeed_gas_price_pb.GasPrice;
      reader.readMessage(value,routerchain_pricefeed_gas_price_pb.GasPrice.deserializeBinaryFromReader);
      msg.addGasPrices(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPriceFeederAddress();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPriceFeederName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getGasPricesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      routerchain_pricefeed_gas_price_pb.GasPrice.serializeBinaryToWriter
    );
  }
};


/**
 * optional string price_feeder_address = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.getPriceFeederAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} returns this
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.setPriceFeederAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string price_feeder_name = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.getPriceFeederName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} returns this
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.setPriceFeederName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * repeated GasPrice gas_prices = 3;
 * @return {!Array<!proto.routerprotocol.routerchain.pricefeed.GasPrice>}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.getGasPricesList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.pricefeed.GasPrice>} */ (
    jspb.Message.getRepeatedWrapperField(this, routerchain_pricefeed_gas_price_pb.GasPrice, 3));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.pricefeed.GasPrice>} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} returns this
*/
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.setGasPricesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.pricefeed.GasPrice=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.pricefeed.GasPrice}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.addGasPrices = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.routerprotocol.routerchain.pricefeed.GasPrice, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPrices} returns this
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPrices.prototype.clearGasPricesList = function() {
  return this.setGasPricesList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse;
  return proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.MsgGasPricesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};


goog.object.extend(exports, proto.routerprotocol.routerchain.pricefeed);
