// @routerprotocol/chain-api/routerchain/pricefeed/update_price_feeder_request_proposal_pb.js
// source: routerchain/pricefeed/update_price_feeder_request_proposal.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var routerchain_pricefeed_price_feeder_info_pb = require('../../routerchain/pricefeed/price_feeder_info_pb.js');
goog.object.extend(proto, routerchain_pricefeed_price_feeder_info_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.displayName = 'proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.toObject = function(includeInstance, msg) {
  var f, obj = {
    title: jspb.Message.getFieldWithDefault(msg, 1, ""),
    description: jspb.Message.getFieldWithDefault(msg, 2, ""),
    priceFeederInfo: (f = msg.getPriceFeederInfo()) && routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal;
  return proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setTitle(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setDescription(value);
      break;
    case 3:
      var value = new routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo;
      reader.readMessage(value,routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo.deserializeBinaryFromReader);
      msg.setPriceFeederInfo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTitle();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getDescription();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getPriceFeederInfo();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo.serializeBinaryToWriter
    );
  }
};


/**
 * optional string title = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.getTitle = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} returns this
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.setTitle = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string description = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.getDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} returns this
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.setDescription = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional PriceFeederInfo price_feeder_info = 3;
 * @return {?proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.getPriceFeederInfo = function() {
  return /** @type{?proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo} */ (
    jspb.Message.getWrapperField(this, routerchain_pricefeed_price_feeder_info_pb.PriceFeederInfo, 3));
};


/**
 * @param {?proto.routerprotocol.routerchain.pricefeed.PriceFeederInfo|undefined} value
 * @return {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} returns this
*/
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.setPriceFeederInfo = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal} returns this
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.clearPriceFeederInfo = function() {
  return this.setPriceFeederInfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.pricefeed.UpdatePriceFeederInfoProposal.prototype.hasPriceFeederInfo = function() {
  return jspb.Message.getField(this, 3) != null;
};


goog.object.extend(exports, proto.routerprotocol.routerchain.pricefeed);
