// @routerprotocol/chain-api/multichain/chain_type_pb.js
// source: multichain/chain_type.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var google_protobuf_any_pb = require('google-protobuf/google/protobuf/any_pb.js');
goog.object.extend(proto, google_protobuf_any_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.multichain.ChainType', null, global);
/**
 * @enum {number}
 */
proto.routerprotocol.routerchain.multichain.ChainType = {
  CHAIN_TYPE_NONE: 0,
  CHAIN_TYPE_ROUTER: 1,
  CHAIN_TYPE_EVM: 2,
  CHAIN_TYPE_COSMOS: 3,
  CHAIN_TYPE_POLKADOT: 4,
  CHAIN_TYPE_SOLANO: 5,
  CHAIN_TYPE_NEAR: 6
};

goog.object.extend(exports, proto.routerprotocol.routerchain.multichain);
