// @routerprotocol/chain-api/attestation/claim_type_pb.js
// source: attestation/claim_type.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var google_protobuf_any_pb = require('google-protobuf/google/protobuf/any_pb.js');
goog.object.extend(proto, google_protobuf_any_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.attestation.ClaimType', null, global);
/**
 * @enum {number}
 */
proto.routerprotocol.routerchain.attestation.ClaimType = {
  CLAIM_TYPE_UNSPECIFIED: 0,
  CLAIM_TYPE_VALSET_UPDATED: 1,
  CLAIM_TYPE_SET_METADATA_REQUEST: 2,
  CLAIM_TYPE_CROSSCHAIN_REQUEST: 3,
  CLAIM_TYPE_CROSSCHAIN_ACK_REQUEST: 4,
  CLAIM_TYPE_CROSSCHAIN_ACK_RECEIPT: 5,
  CLAIM_TYPE_FUND_DEPOSITED_REQUEST: 6,
  CLAIM_TYPE_FUND_PAID_REQUEST: 7
};

goog.object.extend(exports, proto.routerprotocol.routerchain.attestation);
