// @routerprotocol/chain-api/attestation/valset_updated_claim_pb.js
// source: attestation/valset_updated_claim.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var multichain_chain_type_pb = require('../multichain/chain_type_pb.js');
goog.object.extend(proto, multichain_chain_type_pb);
var attestation_bridge_validator_pb = require('../attestation/bridge_validator_pb.js');
goog.object.extend(proto, attestation_bridge_validator_pb);
var attestation_valset_updated_claim_status_pb = require('../attestation/valset_updated_claim_status_pb.js');
goog.object.extend(proto, attestation_valset_updated_claim_status_pb);
var gogoproto_gogo_pb = require('../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.displayName = 'proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.displayName = 'proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.toObject = function(includeInstance, msg) {
  var f, obj = {
    chainid: jspb.Message.getFieldWithDefault(msg, 1, ""),
    contract: jspb.Message.getFieldWithDefault(msg, 2, ""),
    eventnonce: jspb.Message.getFieldWithDefault(msg, 3, 0),
    valsetnonce: jspb.Message.getFieldWithDefault(msg, 4, 0),
    blockheight: jspb.Message.getFieldWithDefault(msg, 5, 0),
    sourcetxhash: jspb.Message.getFieldWithDefault(msg, 6, ""),
    status: jspb.Message.getFieldWithDefault(msg, 7, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim;
  return proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setChainid(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setContract(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventnonce(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setValsetnonce(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBlockheight(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setSourcetxhash(value);
      break;
    case 7:
      var value = /** @type {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimStatus} */ (reader.readEnum());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChainid();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getContract();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getEventnonce();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
  f = message.getValsetnonce();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getBlockheight();
  if (f !== 0) {
    writer.writeUint64(
      5,
      f
    );
  }
  f = message.getSourcetxhash();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getStatus();
  if (f !== 0.0) {
    writer.writeEnum(
      7,
      f
    );
  }
};


/**
 * optional string chainId = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getChainid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setChainid = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string contract = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setContract = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional uint64 eventNonce = 3;
 * @return {number}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getEventnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setEventnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional uint64 valsetNonce = 4;
 * @return {number}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getValsetnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setValsetnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional uint64 blockHeight = 5;
 * @return {number}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getBlockheight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setBlockheight = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};


/**
 * optional string sourceTxHash = 6;
 * @return {string}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getSourcetxhash = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setSourcetxhash = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional ValsetUpdatedClaimStatus status = 7;
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimStatus}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.getStatus = function() {
  return /** @type {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimStatus} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimStatus} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaim.prototype.setStatus = function(value) {
  return jspb.Message.setProto3EnumField(this, 7, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.repeatedFields_ = [6];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.toObject = function(includeInstance, msg) {
  var f, obj = {
    chainid: jspb.Message.getFieldWithDefault(msg, 1, ""),
    contract: jspb.Message.getFieldWithDefault(msg, 2, ""),
    eventnonce: jspb.Message.getFieldWithDefault(msg, 3, 0),
    valsetnonce: jspb.Message.getFieldWithDefault(msg, 4, 0),
    blockheight: jspb.Message.getFieldWithDefault(msg, 5, 0),
    membersList: jspb.Message.toObjectList(msg.getMembersList(),
    attestation_bridge_validator_pb.BridgeValidator.toObject, includeInstance),
    sourcetxhash: jspb.Message.getFieldWithDefault(msg, 7, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash;
  return proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setChainid(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setContract(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventnonce(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setValsetnonce(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBlockheight(value);
      break;
    case 6:
      var value = new attestation_bridge_validator_pb.BridgeValidator;
      reader.readMessage(value,attestation_bridge_validator_pb.BridgeValidator.deserializeBinaryFromReader);
      msg.addMembers(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setSourcetxhash(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChainid();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getContract();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getEventnonce();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
  f = message.getValsetnonce();
  if (f !== 0) {
    writer.writeUint64(
      4,
      f
    );
  }
  f = message.getBlockheight();
  if (f !== 0) {
    writer.writeUint64(
      5,
      f
    );
  }
  f = message.getMembersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      6,
      f,
      attestation_bridge_validator_pb.BridgeValidator.serializeBinaryToWriter
    );
  }
  f = message.getSourcetxhash();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
};


/**
 * optional string chainId = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getChainid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setChainid = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string contract = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getContract = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setContract = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional uint64 eventNonce = 3;
 * @return {number}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getEventnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setEventnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional uint64 valsetNonce = 4;
 * @return {number}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getValsetnonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setValsetnonce = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional uint64 blockHeight = 5;
 * @return {number}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getBlockheight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setBlockheight = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};


/**
 * repeated BridgeValidator members = 6;
 * @return {!Array<!proto.routerprotocol.routerchain.attestation.BridgeValidator>}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getMembersList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.attestation.BridgeValidator>} */ (
    jspb.Message.getRepeatedWrapperField(this, attestation_bridge_validator_pb.BridgeValidator, 6));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.attestation.BridgeValidator>} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
*/
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setMembersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 6, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.attestation.BridgeValidator=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.attestation.BridgeValidator}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.addMembers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 6, opt_value, proto.routerprotocol.routerchain.attestation.BridgeValidator, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.clearMembersList = function() {
  return this.setMembersList([]);
};


/**
 * optional string sourceTxHash = 7;
 * @return {string}
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.getSourcetxhash = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash} returns this
 */
proto.routerprotocol.routerchain.attestation.ValsetUpdatedClaimHash.prototype.setSourcetxhash = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


goog.object.extend(exports, proto.routerprotocol.routerchain.attestation);
