// @routerprotocol/chain-api/metastore/genesis_pb.js
// source: metastore/genesis.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var metastore_params_pb = require('../metastore/params_pb.js');
goog.object.extend(proto, metastore_params_pb);
var metastore_meta_info_pb = require('../metastore/meta_info_pb.js');
goog.object.extend(proto, metastore_meta_info_pb);
var metastore_metadata_request_pb = require('../metastore/metadata_request_pb.js');
goog.object.extend(proto, metastore_metadata_request_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.GenesisState', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.GenesisState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.metastore.GenesisState.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.GenesisState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.GenesisState.displayName = 'proto.routerprotocol.routerchain.metastore.GenesisState';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.metastore.GenesisState.repeatedFields_ = [2,3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.GenesisState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.GenesisState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.GenesisState.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && metastore_params_pb.Params.toObject(includeInstance, f),
    metainfolistList: jspb.Message.toObjectList(msg.getMetainfolistList(),
    metastore_meta_info_pb.MetaInfo.toObject, includeInstance),
    metadatarequestlistList: jspb.Message.toObjectList(msg.getMetadatarequestlistList(),
    metastore_metadata_request_pb.MetadataRequest.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.GenesisState;
  return proto.routerprotocol.routerchain.metastore.GenesisState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.GenesisState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_params_pb.Params;
      reader.readMessage(value,metastore_params_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    case 2:
      var value = new metastore_meta_info_pb.MetaInfo;
      reader.readMessage(value,metastore_meta_info_pb.MetaInfo.deserializeBinaryFromReader);
      msg.addMetainfolist(value);
      break;
    case 3:
      var value = new metastore_metadata_request_pb.MetadataRequest;
      reader.readMessage(value,metastore_metadata_request_pb.MetadataRequest.deserializeBinaryFromReader);
      msg.addMetadatarequestlist(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.GenesisState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.GenesisState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.GenesisState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      metastore_params_pb.Params.serializeBinaryToWriter
    );
  }
  f = message.getMetainfolistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      metastore_meta_info_pb.MetaInfo.serializeBinaryToWriter
    );
  }
  f = message.getMetadatarequestlistList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      metastore_metadata_request_pb.MetadataRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.routerprotocol.routerchain.metastore.Params}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.getParams = function() {
  return /** @type{?proto.routerprotocol.routerchain.metastore.Params} */ (
    jspb.Message.getWrapperField(this, metastore_params_pb.Params, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.metastore.Params|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState} returns this
*/
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState} returns this
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated MetaInfo metaInfoList = 2;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.getMetainfolistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_meta_info_pb.MetaInfo, 2));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} value
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState} returns this
*/
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.setMetainfolistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetaInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetaInfo}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.addMetainfolist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.routerprotocol.routerchain.metastore.MetaInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState} returns this
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.clearMetainfolistList = function() {
  return this.setMetainfolistList([]);
};


/**
 * repeated MetadataRequest metadataRequestList = 3;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetadataRequest>}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.getMetadatarequestlistList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetadataRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_metadata_request_pb.MetadataRequest, 3));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetadataRequest>} value
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState} returns this
*/
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.setMetadatarequestlistList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetadataRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetadataRequest}
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.addMetadatarequestlist = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.routerprotocol.routerchain.metastore.MetadataRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.GenesisState} returns this
 */
proto.routerprotocol.routerchain.metastore.GenesisState.prototype.clearMetadatarequestlistList = function() {
  return this.setMetadatarequestlistList([]);
};


goog.object.extend(exports, proto.routerprotocol.routerchain.metastore);
