// @routerprotocol/chain-api/metastore/query_pb.js
// source: metastore/query.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() { return this || window || global || self || Function('return this')(); }).call(null);

var gogoproto_gogo_pb = require('../gogoproto/gogo_pb.js');
goog.object.extend(proto, gogoproto_gogo_pb);
var google_api_annotations_pb = require('../google/api/annotations_pb.js');
goog.object.extend(proto, google_api_annotations_pb);
var cosmos_base_query_v1beta1_pagination_pb = require('../cosmos/base/query/v1beta1/pagination_pb.js');
goog.object.extend(proto, cosmos_base_query_v1beta1_pagination_pb);
var metastore_params_pb = require('../metastore/params_pb.js');
goog.object.extend(proto, metastore_params_pb);
var metastore_meta_info_pb = require('../metastore/meta_info_pb.js');
goog.object.extend(proto, metastore_meta_info_pb);
var metastore_metadata_request_pb = require('../metastore/metadata_request_pb.js');
goog.object.extend(proto, metastore_metadata_request_pb);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.MetaInfoQueryStatus', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryParamsRequest', null, global);
goog.exportSymbol('proto.routerprotocol.routerchain.metastore.QueryParamsResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryParamsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryParamsRequest.displayName = 'proto.routerprotocol.routerchain.metastore.QueryParamsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryParamsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryParamsResponse.displayName = 'proto.routerprotocol.routerchain.metastore.QueryParamsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.displayName = 'proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.displayName = 'proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.displayName = 'proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.displayName = 'proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.repeatedFields_, null);
};
goog.inherits(proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.displayName = 'proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryParamsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryParamsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryParamsRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryParamsRequest;
  return proto.routerprotocol.routerchain.metastore.QueryParamsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryParamsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryParamsRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryParamsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryParamsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryParamsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryParamsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    params: (f = msg.getParams()) && metastore_params_pb.Params.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryParamsResponse;
  return proto.routerprotocol.routerchain.metastore.QueryParamsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_params_pb.Params;
      reader.readMessage(value,metastore_params_pb.Params.deserializeBinaryFromReader);
      msg.setParams(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryParamsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParams();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      metastore_params_pb.Params.serializeBinaryToWriter
    );
  }
};


/**
 * optional Params params = 1;
 * @return {?proto.routerprotocol.routerchain.metastore.Params}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.prototype.getParams = function() {
  return /** @type{?proto.routerprotocol.routerchain.metastore.Params} */ (
    jspb.Message.getWrapperField(this, metastore_params_pb.Params, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.metastore.Params|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.prototype.setParams = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryParamsResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.prototype.clearParams = function() {
  return this.setParams(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryParamsResponse.prototype.hasParams = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    chainId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    address: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest;
  return proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setChainId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddress(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChainId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getAddress();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string chain_id = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.prototype.getChainId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.prototype.setChainId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string address = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.prototype.getAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoRequest.prototype.setAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    metainfo: (f = msg.getMetainfo()) && metastore_meta_info_pb.MetaInfo.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse;
  return proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_meta_info_pb.MetaInfo;
      reader.readMessage(value,metastore_meta_info_pb.MetaInfo.deserializeBinaryFromReader);
      msg.setMetainfo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetainfo();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      metastore_meta_info_pb.MetaInfo.serializeBinaryToWriter
    );
  }
};


/**
 * optional MetaInfo metaInfo = 1;
 * @return {?proto.routerprotocol.routerchain.metastore.MetaInfo}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.prototype.getMetainfo = function() {
  return /** @type{?proto.routerprotocol.routerchain.metastore.MetaInfo} */ (
    jspb.Message.getWrapperField(this, metastore_meta_info_pb.MetaInfo, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.metastore.MetaInfo|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.prototype.setMetainfo = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.prototype.clearMetainfo = function() {
  return this.setMetainfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetaInfoResponse.prototype.hasMetainfo = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.toObject = function(includeInstance, msg) {
  var f, obj = {
    chainIds: jspb.Message.getFieldWithDefault(msg, 1, ""),
    address: jspb.Message.getFieldWithDefault(msg, 2, ""),
    status: jspb.Message.getFieldWithDefault(msg, 3, 0),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageRequest.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setChainIds(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddress(value);
      break;
    case 3:
      var value = /** @type {!proto.routerprotocol.routerchain.metastore.MetaInfoQueryStatus} */ (reader.readEnum());
      msg.setStatus(value);
      break;
    case 4:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageRequest;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageRequest.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChainIds();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getAddress();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getStatus();
  if (f !== 0.0) {
    writer.writeEnum(
      3,
      f
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional string chain_ids = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.getChainIds = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.setChainIds = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string address = 2;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.getAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.setAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional MetaInfoQueryStatus status = 3;
 * @return {!proto.routerprotocol.routerchain.metastore.MetaInfoQueryStatus}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.getStatus = function() {
  return /** @type {!proto.routerprotocol.routerchain.metastore.MetaInfoQueryStatus} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetaInfoQueryStatus} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.setStatus = function(value) {
  return jspb.Message.setProto3EnumField(this, 3, value);
};


/**
 * optional cosmos.base.query.v1beta1.PageRequest pagination = 4;
 * @return {?proto.cosmos.base.query.v1beta1.PageRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageRequest} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageRequest, 4));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageRequest|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainAndAddress.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 4) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.toObject = function(includeInstance, msg) {
  var f, obj = {
    metainfoList: jspb.Message.toObjectList(msg.getMetainfoList(),
    metastore_meta_info_pb.MetaInfo.toObject, includeInstance),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageResponse.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_meta_info_pb.MetaInfo;
      reader.readMessage(value,metastore_meta_info_pb.MetaInfo.deserializeBinaryFromReader);
      msg.addMetainfo(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageResponse;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageResponse.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetainfoList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      metastore_meta_info_pb.MetaInfo.serializeBinaryToWriter
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageResponse.serializeBinaryToWriter
    );
  }
};


/**
 * repeated MetaInfo metaInfo = 1;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.getMetainfoList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_meta_info_pb.MetaInfo, 1));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.setMetainfoList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetaInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetaInfo}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.addMetainfo = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.routerprotocol.routerchain.metastore.MetaInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.clearMetainfoList = function() {
  return this.setMetainfoList([]);
};


/**
 * optional cosmos.base.query.v1beta1.PageResponse pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageResponse} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageResponse, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageResponse|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainAndAddress.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageRequest.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageRequest;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageRequest.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional cosmos.base.query.v1beta1.PageRequest pagination = 1;
 * @return {?proto.cosmos.base.query.v1beta1.PageRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageRequest} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageRequest, 1));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageRequest|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequest.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 1) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    metainfoList: jspb.Message.toObjectList(msg.getMetainfoList(),
    metastore_meta_info_pb.MetaInfo.toObject, includeInstance),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageResponse.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_meta_info_pb.MetaInfo;
      reader.readMessage(value,metastore_meta_info_pb.MetaInfo.deserializeBinaryFromReader);
      msg.addMetainfo(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageResponse;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageResponse.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetainfoList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      metastore_meta_info_pb.MetaInfo.serializeBinaryToWriter
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageResponse.serializeBinaryToWriter
    );
  }
};


/**
 * repeated MetaInfo metaInfo = 1;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.getMetainfoList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_meta_info_pb.MetaInfo, 1));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.setMetainfoList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetaInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetaInfo}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.addMetainfo = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.routerprotocol.routerchain.metastore.MetaInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.clearMetainfoList = function() {
  return this.setMetainfoList([]);
};


/**
 * optional cosmos.base.query.v1beta1.PageResponse pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageResponse} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageResponse, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageResponse|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponse.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    chainId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    eventNonce: jspb.Message.getFieldWithDefault(msg, 3, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest;
  return proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setChainId(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEventNonce(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChainId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getEventNonce();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
};


/**
 * optional string chain_id = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.prototype.getChainId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.prototype.setChainId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional uint64 event_nonce = 3;
 * @return {number}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.prototype.getEventNonce = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestRequest.prototype.setEventNonce = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    metadatarequest: (f = msg.getMetadatarequest()) && metastore_metadata_request_pb.MetadataRequest.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse;
  return proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_metadata_request_pb.MetadataRequest;
      reader.readMessage(value,metastore_metadata_request_pb.MetadataRequest.deserializeBinaryFromReader);
      msg.setMetadatarequest(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetadatarequest();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      metastore_metadata_request_pb.MetadataRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional MetadataRequest metadataRequest = 1;
 * @return {?proto.routerprotocol.routerchain.metastore.MetadataRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.prototype.getMetadatarequest = function() {
  return /** @type{?proto.routerprotocol.routerchain.metastore.MetadataRequest} */ (
    jspb.Message.getWrapperField(this, metastore_metadata_request_pb.MetadataRequest, 1));
};


/**
 * @param {?proto.routerprotocol.routerchain.metastore.MetadataRequest|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.prototype.setMetadatarequest = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.prototype.clearMetadatarequest = function() {
  return this.setMetadatarequest(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryGetMetadataRequestResponse.prototype.hasMetadatarequest = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageRequest.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageRequest;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageRequest.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional cosmos.base.query.v1beta1.PageRequest pagination = 1;
 * @return {?proto.cosmos.base.query.v1beta1.PageRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageRequest} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageRequest, 1));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageRequest|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestRequest.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 1) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    metadatarequestList: jspb.Message.toObjectList(msg.getMetadatarequestList(),
    metastore_metadata_request_pb.MetadataRequest.toObject, includeInstance),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageResponse.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_metadata_request_pb.MetadataRequest;
      reader.readMessage(value,metastore_metadata_request_pb.MetadataRequest.deserializeBinaryFromReader);
      msg.addMetadatarequest(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageResponse;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageResponse.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetadatarequestList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      metastore_metadata_request_pb.MetadataRequest.serializeBinaryToWriter
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageResponse.serializeBinaryToWriter
    );
  }
};


/**
 * repeated MetadataRequest metadataRequest = 1;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetadataRequest>}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.getMetadatarequestList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetadataRequest>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_metadata_request_pb.MetadataRequest, 1));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetadataRequest>} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.setMetadatarequestList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetadataRequest=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetadataRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.addMetadatarequest = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.routerprotocol.routerchain.metastore.MetadataRequest, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.clearMetadatarequestList = function() {
  return this.setMetadatarequestList([]);
};


/**
 * optional cosmos.base.query.v1beta1.PageResponse pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageResponse} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageResponse, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageResponse|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetadataRequestResponse.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.toObject = function(includeInstance, msg) {
  var f, obj = {
    address: jspb.Message.getFieldWithDefault(msg, 1, ""),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageRequest.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setAddress(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageRequest;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageRequest.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAddress();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional string address = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.getAddress = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.setAddress = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional cosmos.base.query.v1beta1.PageRequest pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageRequest} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageRequest, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageRequest|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByAddress.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.toObject = function(includeInstance, msg) {
  var f, obj = {
    metainfoList: jspb.Message.toObjectList(msg.getMetainfoList(),
    metastore_meta_info_pb.MetaInfo.toObject, includeInstance),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageResponse.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_meta_info_pb.MetaInfo;
      reader.readMessage(value,metastore_meta_info_pb.MetaInfo.deserializeBinaryFromReader);
      msg.addMetainfo(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageResponse;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageResponse.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetainfoList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      metastore_meta_info_pb.MetaInfo.serializeBinaryToWriter
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageResponse.serializeBinaryToWriter
    );
  }
};


/**
 * repeated MetaInfo metaInfo = 1;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.getMetainfoList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_meta_info_pb.MetaInfo, 1));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.setMetainfoList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetaInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetaInfo}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.addMetainfo = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.routerprotocol.routerchain.metastore.MetaInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.clearMetainfoList = function() {
  return this.setMetainfoList([]);
};


/**
 * optional cosmos.base.query.v1beta1.PageResponse pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageResponse} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageResponse, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageResponse|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByAddress.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.toObject = function(includeInstance, msg) {
  var f, obj = {
    chainIds: jspb.Message.getFieldWithDefault(msg, 1, ""),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageRequest.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setChainIds(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageRequest;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageRequest.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChainIds();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageRequest.serializeBinaryToWriter
    );
  }
};


/**
 * optional string chain_ids = 1;
 * @return {string}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.getChainIds = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.setChainIds = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional cosmos.base.query.v1beta1.PageRequest pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageRequest}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageRequest} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageRequest, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageRequest|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoRequestByChainID.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.toObject = function(opt_includeInstance) {
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.toObject = function(includeInstance, msg) {
  var f, obj = {
    metainfoList: jspb.Message.toObjectList(msg.getMetainfoList(),
    metastore_meta_info_pb.MetaInfo.toObject, includeInstance),
    pagination: (f = msg.getPagination()) && cosmos_base_query_v1beta1_pagination_pb.PageResponse.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID;
  return proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new metastore_meta_info_pb.MetaInfo;
      reader.readMessage(value,metastore_meta_info_pb.MetaInfo.deserializeBinaryFromReader);
      msg.addMetainfo(value);
      break;
    case 2:
      var value = new cosmos_base_query_v1beta1_pagination_pb.PageResponse;
      reader.readMessage(value,cosmos_base_query_v1beta1_pagination_pb.PageResponse.deserializeBinaryFromReader);
      msg.setPagination(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMetainfoList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      metastore_meta_info_pb.MetaInfo.serializeBinaryToWriter
    );
  }
  f = message.getPagination();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      cosmos_base_query_v1beta1_pagination_pb.PageResponse.serializeBinaryToWriter
    );
  }
};


/**
 * repeated MetaInfo metaInfo = 1;
 * @return {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.getMetainfoList = function() {
  return /** @type{!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} */ (
    jspb.Message.getRepeatedWrapperField(this, metastore_meta_info_pb.MetaInfo, 1));
};


/**
 * @param {!Array<!proto.routerprotocol.routerchain.metastore.MetaInfo>} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.setMetainfoList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.routerprotocol.routerchain.metastore.MetaInfo=} opt_value
 * @param {number=} opt_index
 * @return {!proto.routerprotocol.routerchain.metastore.MetaInfo}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.addMetainfo = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.routerprotocol.routerchain.metastore.MetaInfo, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.clearMetainfoList = function() {
  return this.setMetainfoList([]);
};


/**
 * optional cosmos.base.query.v1beta1.PageResponse pagination = 2;
 * @return {?proto.cosmos.base.query.v1beta1.PageResponse}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.getPagination = function() {
  return /** @type{?proto.cosmos.base.query.v1beta1.PageResponse} */ (
    jspb.Message.getWrapperField(this, cosmos_base_query_v1beta1_pagination_pb.PageResponse, 2));
};


/**
 * @param {?proto.cosmos.base.query.v1beta1.PageResponse|undefined} value
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} returns this
*/
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.setPagination = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID} returns this
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.clearPagination = function() {
  return this.setPagination(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.routerprotocol.routerchain.metastore.QueryAllMetaInfoResponseByChainID.prototype.hasPagination = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * @enum {number}
 */
proto.routerprotocol.routerchain.metastore.MetaInfoQueryStatus = {
  ALL: 0,
  PENDING: 1,
  APPROVED: 2
};

goog.object.extend(exports, proto.routerprotocol.routerchain.metastore);
