// @routerprotocol/chain-api/metastore/tx_pb_service.js
// package: routerprotocol.routerchain.metastore
// file: metastore/tx.proto

var metastore_tx_pb = require("../metastore/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "routerprotocol.routerchain.metastore.Msg";
  return Msg;
}());

Msg.CreateMetadataRequest = {
  methodName: "CreateMetadataRequest",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: metastore_tx_pb.MsgCreateMetadataRequest,
  responseType: metastore_tx_pb.MsgCreateMetadataRequestResponse
};

Msg.ApproveFeepayerRequest = {
  methodName: "ApproveFeepayerRequest",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: metastore_tx_pb.MsgApproveFeepayerRequest,
  responseType: metastore_tx_pb.MsgApproveFeepayerRequestResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.createMetadataRequest = function createMetadataRequest(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.CreateMetadataRequest, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

MsgClient.prototype.approveFeepayerRequest = function approveFeepayerRequest(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.ApproveFeepayerRequest, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
